prometheus/promhttp/http.go

```diff
@@ -346,6 +376,9 @@ type HandlerOpts struct {
 	// If DisableCompression is true, the handler will never compress the
 	// response, even if requested by the client.
 	DisableCompression bool
+	// If DisableCompression is false, this option will allow to define the
+	// set of offered encoding algorithms.
+	EncodingOffers []string
```

Comment:
I feel we should have a better name than "EncodingOffers". Offers is bit weird name (it makes sense from negotiation library POV), but maybe is better than "Accepted". "Encoding" is weird as we used "Compression" in the first field. There is also potential overlap functionality with DisableCompression. Quickly thinking.. maybe we should clarify all (this also proposes error handling)

```suggestion
	// DisableCompression disables the response encoding (compression) and
	// encoding negotiation. If true, the handler will 
	// never compress the response, even if requested
	// by the client and the OfferedCompressions field is ignored.
	DisableCompression bool
	// OfferedCompressions is a set of encodings (compressions) handler will
	// try to offer when negotiating with the client. This defaults to zstd, gzip, identity
	//
	// NOTE: If handler can't agree on the encodings with the client or caller using unsupported or empty encodings in OfferedCompressions, handler always fallbacks to
	// no compression (identity), for compatibility reasons. In such cases ErrorLog will be used if set.
        OfferedCompressions []encoding
```